encodings/dict/src/array.rs

```diff
@@ -257,4 +283,56 @@ mod test {
         };
         assert_eq!(indices, [2, 4]);
     }
+
+    fn make_dict_primitive_chunks<T: NativePType, U: NativePType>(
```

Comment:
I think we have a logic to generate random dictionaries in 3 places. Might make sense to somehow share it but this is probably for later

Reply:
I agree, I think there is lots duplicity in our benchmarks